### PR TITLE
Use structured logging in gerrit packages.

### PR DESCRIPTION
**Pull request by fejta**
This should make it easier to search for and debug logs.

Also reduce some indentation by continuing/returning early.

### CODE REVIEW

File: prow/gerrit/client/client.go [hunk 1]
```diff
@@ -246,11 +257,12 @@ func (c *Client) QueryChanges(lastState LastSyncState, rateLimit int) map[string
...
-			result[h.instance] = []ChangeInfo{}
-			for _, change := range changes {
-				result[h.instance] = append(result[h.instance], change)
-			}
+		if len(changes) == 0 {
+			continue
+		}
+
+		for _, change := range changes {
+			result[h.instance] = append(result[h.instance], change)
```
Comment (chizhg) [inline prow/gerrit/client/client.go:265]: Is it safe to assume `h.instance` always exists in the `result` map? In the deleted code there is `result[h.instance] = []ChangeInfo{}`

File: prow/gerrit/client/client.go [hunk 2]
```diff
@@ -366,72 +382,86 @@ func (h *gerritInstanceHandler) queryChangesForProject(project string, lastUpdat
...
-					// change has been abandoned, do nothing
+				if !newMessages && !created.After(lastUpdate) {
+					// stale commit
+					log.Info("Skipping existing change")
+					continue
 				}
-			} else {
-				logrus.Infof("Change %d, updated %s before lastUpdate %s, return", change.Number, change.Updated, lastUpdate)
-				return pending, nil
+				if !newMessages {
```
Comment (chizhg) [inline prow/gerrit/client/client.go:458]: The log message is opposite from what's meant by this if check, should it be `if newMessages`?

File: prow/gerrit/client/client.go (hunk 1 above)
Comment (fejta) [reply prow/gerrit/client/client.go:265]: That is not necessary.

```golang
var foo []string = nil
foo = append(foo, "hello")
```

works just fine. This is in fact one of the reasons why append returns a value.

See https://play.golang.org/p/EVcqaQ0JNOI

File: prow/gerrit/client/client.go (hunk 2 above)
Comment (fejta) [reply prow/gerrit/client/client.go:458]: If there are new messages then we got to L447 which means we already logged on line L443.

L443 covers the case where the change was updated by a comment.

L459 covers the case where someone pushed new code.

Comment (chizhg) [reply prow/gerrit/client/client.go:458]: Got it. Thanks for the explanation!

File: prow/gerrit/client/client.go (hunk 1 above)
Comment (chizhg) [reply prow/gerrit/client/client.go:265]: I see. Thanks for the explanation and example!

**Pull request merged**